transform_encoder.py

```diff
@@ -0,0 +1,131 @@
+        base_tokenizer_model: Optional[str] = None,
+        pooling_strategy: str = "mean",
+        layer_index: int = -1,
+        max_length: Optional[int] = None,
+        embedding_fn_name: str = "__call__",
+        device: str = "cpu",
+        default_traversal_path: str = "r",
+        default_batch_size: int = 32,
+        *args,
+        **kwargs,
+    ):
+        """
+        :param pretrained_model_name_or_path:
+        :param base_tokenizer_model:
+        :param pooling_strategy:
+        :param layer_index:
+        :param max_length:
+        :param embedding_fn_name:
+        :param device:
+        :param default_traversal_path: Used in the encode method an define traversal on the received `DocumentArray`
+        :param default_batch_size: Defines the batch size for inference on the loaded PyTorch model.
+        :param args:
+        :param kwargs:
+        """
+        super().__init__(*args, **kwargs)
+        self.default_traversal_path = default_traversal_path
+        self.default_batch_size = default_batch_size
+        self.pretrained_model_name_or_path = pretrained_model_name_or_path
+        self.base_tokenizer_model = (
+            base_tokenizer_model or pretrained_model_name_or_path
+        )
+        self.pooling_strategy = pooling_strategy
+        self.layer_index = layer_index
+        self.max_length = max_length
+        if not device in ["cpu", "cuda"]:
+            logger.error("Torch device not supported. Must be cpu or cuda!")
+            raise RuntimeError("Torch device not supported. Must be cpu or cuda!")
+        if device == "cuda" and not torch.cuda.is_available():
+            logger.warning(
+                "You tried to use GPU but torch did not detect your"
+                "GPU correctly. Defaulting to CPU. Check your CUDA installation!"
+            )
+            device = "cpu"
+        self.device = device
+        self.embedding_fn_name = embedding_fn_name
+        self.tokenizer = AutoTokenizer.from_pretrained(self.base_tokenizer_model)
+        self.model = AutoModel.from_pretrained(
+            self.pretrained_model_name_or_path, output_hidden_states=True
+        )
+        self.model.to(torch.device(device))
+
+    def _compute_embedding(
+        self, hidden_states: Tuple["torch.Tensor"], input_tokens: Dict
+    ):
+        fill_vals = {"cls": 0.0, "mean": 0.0, "max": -np.inf, "min": np.inf}
+        fill_val = torch.tensor(
+            fill_vals[self.pooling_strategy], device=torch.device(self.device)
+        )
+        layer = hidden_states[self.layer_index]
+        attn_mask = input_tokens["attention_mask"].unsqueeze(-1).expand_as(layer)
+        layer = torch.where(attn_mask.bool(), layer, fill_val)
+        embeddings = layer.sum(dim=1) / attn_mask.sum(dim=1)
+        return embeddings.cpu().numpy()
+
+    @requests
+    def encode(self, docs: "DocumentArray", parameters: Dict, **kwargs):
```

Comment:
DO not need quotes

Reply:
https://github.com/jina-ai/executor-transformer-torch-encoder/pull/1/commits/9116b7c45458db6495de5eca7b8f2d31dd4104c1